### PR TITLE
15 tk 016create page to enter sms otp

### PR DESCRIPTION
**Pull request by NkwaTambe**
### Title: Create SMS OTP Entry Page
Documentation Impl : https://dominicarrojado.com/posts/how-to-create-your-own-otp-input-in-react-and-typescript-with-tests-part-1/

Documentation Test : https://dominicarrojado.com/posts/how-to-create-your-own-otp-input-in-react-and-typescript-with-tests-part-2/

### Description:
This pull request introduces a new page for users to enter their SMS-based One-Time Password (OTP) during the authentication process. The implementation includes:

An intuitive layout with input fields for a 4-digit OTP.
Automatic focus management between input fields for enhanced user experience.
Validation to ensure only numeric input is accepted.
A resend OTP feature for user convenience.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -4,6 +4,7 @@
     <meta charset="UTF-8" />
     
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link href="./src/output.css" rel="stylesheet">
     <title>webank-UserApp + TS</title>
```
Comment (Arielpetit) [inline index.html:8]: Hello @NkwaTambe Please change this title to Webank-UserApp

**Review by Arielpetit — COMMENTED** [review]
I think a test is missing, Please do well to add it

**Review by Arielpetit — APPROVED** [review]

File: src/components/OtpInput.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      // If the character matches a digit, use it, otherwise use an empty string
+      if (RE_DIGIT.test(char)) {
+        items.push(char);
+      } else {
+        items.push("");
+      }
+    }
+
+    return items;
+  }, [value, valueLength]);
```
Comment (Arielpetit) [inline src/components/OtpInput.tsx:30]: Please put that in a proper method with a method name and a unit test for that

**Review by Arielpetit — CHANGES_REQUESTED** [review]

File: src/components/OtpInput.tsx (hunk 2 above)
Comment (NkwaTambe) [reply src/components/OtpInput.tsx:30]: done

**Pull request merged**